--- jdk/make/common/Defs.gmk	Thu Jan 16 11:52:13 2014 -0800
+++ jdk/make/common/Defs.gmk	Fri Jan 17 14:36:07 2014 +0100
@@ -180,7 +180,7 @@
   ifdef ALT_FREETYPE_LIB_PATH
     FREETYPE_LIB_PATH = $(ALT_FREETYPE_LIB_PATH)
     ifeq ($(PLATFORM), macosx)
-      USING_SYSTEM_FT_LIB=true
+      USING_SYSTEM_FT_LIB=false
     endif
   else
     ifeq ($(DEVTOOLS_FT_DIR_EXISTS), true)
--- jdk/make/sun/font/Makefile	Thu Jan 16 11:52:13 2014 -0800
+++ jdk/make/sun/font/Makefile	Fri Jan 17 14:36:07 2014 +0100
@@ -136,15 +136,24 @@
     OTHER_LDLIBS += $(FREETYPE_LIB_PATH)/freetype.lib
 else 
   ifeq ($(USING_SYSTEM_FT_LIB), false)
-    FREETYPE_LIB = $(LIB_LOCATION)/$(LIB_PREFIX)freetype.$(LIBRARY_SUFFIX).6
+    ifeq ($(PLATFORM), macosx)
+      FREETYPE_LIB = $(LIB_LOCATION)/$(LIB_PREFIX)freetype.$(LIBRARY_SUFFIX)
+      OTHER_LDLIBS += -lfreetype
+    else
+      FREETYPE_LIB = $(LIB_LOCATION)/$(LIB_PREFIX)freetype.$(LIBRARY_SUFFIX).6
+      OTHER_LDLIBS += -L$(FREETYPE_LIB_PATH) -lfreetype
+    endif
   endif
-  OTHER_LDLIBS += -L$(FREETYPE_LIB_PATH) -lfreetype
 endif 
 
 library:: $(FREETYPE_LIB)
 
 $(FREETYPE_LIB):
 	$(CP) $(FREETYPE_LIB_PATH)/$(LIB_PREFIX)freetype.$(LIBRARY_SUFFIX) $@
+
+  ifeq ($(PLATFORM), macosx)
+    install_name_tool -id @rpath/libfreetype.dylib $@
+  endif
 endif
 
 #ifeq ($(PLATFORM), solaris)
